templates/admin/asset/index.html.ep

```diff
@@ -16,96 +16,78 @@
+                    % if ($asset->{max_job}) {
+                       %= link_to($asset->{max_job} => url_for('test', testid => $asset->{max_job}))
+                    % } else {
+                      0
+                    % }
                 </td>
-                <td class="size"><%= $asset->size %></td>
+                <td class="size"><%= $asset->{size} %></td>
                 <td class="last_use">
-                    % my $last_use_job = $asset->last_use_job;
-                    % if (my $group = $last_use_job->group) {
-                        <%= link_to($group->name => url_for(is_admin() ? 'admin_job_templates' : 'group_overview', groupid => $group->id)); %>:
-                    % }
-                    %= link_to($last_use_job->name => url_for('test', testid => $last_use_job->id))
+                   % for my $group (sort { $a <=> $b } keys(%{$asset->{groups}})) {
+                     % my $class = 'not-picked';
+                     % if ($asset->{picked_into} == $group) {
+                       % $class = 'picked-group';
+                     % }
+                     % if ($asset->{picked_into} == 0) {
+                       % $class = 'to-be-removed';
+                     % }
+                     %=  link_to $group => url_for('test', testid => $asset->{groups}->{$group}) => (class => $class)
+                   % }
                 </td>
             </tr>
             % }
-
             </tbody>
         </table>
 
-        <h3>Untracked assets</h3>
-        <p>The following assets are not used by any job or only by jobs which are not part of any group.</p>
-        <table id="untracked-assets" class="display table table-striped">
-            <thead>
-                <tr>
-                    <th>Type</th>
-                    <th>Name</th>
-                    <th>#Jobs</th>
-                    <th>Created</th>
-                    <th>Size</th>
-                </tr>
-            </thead>
-            <tbody>
-            % $assets->reset();
-            % while (my $asset = $assets->next()) {
-                % next if ($asset->last_use_job_id);
-                <tr id="asset_<%= $asset->id %>">
-                    <td class="type"><%= $asset->type %></td>
-                    <td class="name">
-                        %= $asset->name
-                        <a href="#" onclick="deleteAsset(<%= $asset->id %>);">
-                            <i class="action far fa-fw fa-times-circle" title="Delete asset"></i>
-                        </a>
-                    </td>
-                    <td class="nrjobs"><%= link_to($asset->jobs_assets->count => url_for('tests')->query(assetid => $asset->id)) %></td>
-                    <td class="t_created">
-                        %= $asset->t_created
-                    </td>
-                    <td class="size"><%= $asset->size %></td>
-                </tr>
-            % }
-            </tbody>
-        </table>
+        % my $sum = 0;
```

Comment:
That kind of logic should be in the controller, not in the view. Or you make it an one-liner (maybe Perl has something like `std::accumulate`).

Reply:
I can't say I find it easier to parse, but the one line version is:
```
my $total_size = sum(map { $_->{picked} } values(%{$status->{groups}}));
```